crates/namada/src/vm/host_env.rs

```diff
@@ -2072,7 +2073,11 @@ where
 {
     let code_hash = Hash::try_from(code_hash)
         .map_err(|e| TxRuntimeError::InvalidVpCodeHash(e.to_string()))?;
-    let state = env.state();
+
+    // NB: an env clone is required to avoid an immutable
+    // borrow while a mutable borrow is taking place
+    let env2 = env.clone();
+    let state = env2.state();
```

Comment:
I think we can inline `env.state()` below without binding it to a `let` to avoid this